alot: 0.9.1 -> 0.10

lucc opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

New release of alot: https://github.com/pazz/alot/releases/tag/0.10

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@edibopp please review.

lucc committed:
python3Packages.alot: 0.9.1 -> 0.10

fabaff reviewed (commented):
Please update the commit message. It's located in the Python modules area.

```bash
python3Packages.alot: 0.9.1 -> 0.10
```

Could you also update the license please?  It's `gpl3Plus` (-> https://github.com/pazz/alot/blob/master/setup.py#L22 and the file headers).

lucc commented:
@fabaff fixed.

Does anybody know how I can exclude certain unittests? alot uses `python setup.py test` with the standard lib unitests module from python.  There are exactly two tests that fail in NixOS' sandbox:
```
======================================================================
FAIL: test_no_spawn_no_stdin_attached (tests.commands.test_global.TestExternalCommand)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/tests/utilities.py", line 189, in _actual
    return loop.run_until_complete(coro(*args, **kwargs))
  File "/nix/store/i1m8r7mv8h47wr850cdsxksy22lv6gsz-python3-3.9.6/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/build/source/tests/commands/test_global.py", line 129, in test_no_spawn_no_stdin_attached
    ui.notify.assert_not_called()
  File "/nix/store/i1m8r7mv8h47wr850cdsxksy22lv6gsz-python3-3.9.6/lib/python3.9/unittest/mock.py", line 868, in assert_not_called
    raise AssertionError(msg)
AssertionError: Expected 'notify' to not have been called. Called 1 times.
Calls: [call('editor has exited with error code 1 -- No stderr output', priority='error')].

======================================================================
FAIL: test_env_set (tests.test_helper.TestCallCmdAsync)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/source/tests/utilities.py", line 189, in _actual
    return loop.run_until_complete(coro(*args, **kwargs))
  File "/nix/store/i1m8r7mv8h47wr850cdsxksy22lv6gsz-python3-3.9.6/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/build/source/tests/test_helper.py", line 433, in test_env_set
    self.assertEqual(ret[0], 'bar')
AssertionError: '' != 'bar'
+ bar
```
 and because of this we have `doCheck=false` in the alot derivation.  I would like to deactivate these two and run the remaining tests.

lucc committed:
python3Packages.alot: exclude some tests but run the test suite

Some test are broken in the Nix sandbox.  This way it is possible to at
least run the test suite and catch errors from the other tests.

This approach is documented in
https://github.com/NixOS/nixpkgs/blob/d6c2bb7/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

lucc commented:
I marked the two problematic tests as expected failure. This is now ready for review (and merge from my point of view).

Artturin commented:
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

lucc commented:
@Artturin thanks for the hint that is much nicer.

SuperSandro2000 commented on pkgs/development/python-modules/alot/default.nix:
```diff
@@ -35,11 +35,18 @@ buildPythonPackage rec {
   postBuild = lib.optionalString withManpage "make -C docs man";
 
-  checkInputs =  [ gawk future mock gnupg procps ];
+  doCheck = true;
```
```suggestion
```
default

SuperSandro2000 commented on pkgs/development/python-modules/alot/default.nix:
```diff
@@ -35,11 +35,18 @@ buildPythonPackage rec {
 
-  checkInputs =  [ gawk future mock gnupg procps ];
+  doCheck = true;
+  checkInputs = [ gawk future mock gnupg procps pytestCheckHook ];
+  #checkPhase = "pytest";
```
```suggestion
  checkInputs = [ gawk future mock gnupg procps pytestCheckHook ];

```

SuperSandro2000 commented on pkgs/development/python-modules/alot/default.nix:
```diff
@@ -35,11 +35,18 @@ buildPythonPackage rec {
+  disabledTests = [
+    "test_env_set"
+    "test_no_spawn_no_stdin_attached"
+     #"tests/test_helper.py::TestCallCmdAsync::test_env_set"
+     #"tests/commands/test_global.py::TestExternalCommand::test_no_spawn_no_stdin_attached"
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/alot/default.nix:
```diff
@@ -35,11 +35,18 @@ buildPythonPackage rec {
+  doCheck = true;
+  checkInputs = [ gawk future mock gnupg procps pytestCheckHook ];
+  #checkPhase = "pytest";
+  # some twisted tests need the network (test_env_set... )
```
```suggestion
  # some twisted tests need internet access
```

lucc commented:
@SuperSandro2000 resolved.
